crt_portal/cts_forms/filters.py

```diff
@@ -62,10 +63,31 @@ def report_filter(request):
                 kwargs[f'{field}__icontains'] = request.GET.getlist(field)[0]
             elif 'date' in field:
                 # filters by a start date or an end date expects YYYYMMDD
+                # year = int(request.GET.getlist(field)[0][:4])
+                # month = int(request.GET.getlist(field)[0][4:6])
+                # day = int(request.GET.getlist(field)[0][6:])
+                # filters by a start date or an end date expects yyyy-mm-dd, or yyyy-mm, or yyyy
                 field_name = _get_date_field_from_param(field)
-                year = int(request.GET.getlist(field)[0][:4])
-                month = int(request.GET.getlist(field)[0][4:6])
-                day = int(request.GET.getlist(field)[0][6:])
+                encodedDate = request.GET.getlist(field)[0]
+                decodedDate = urllib.parse.unquote(encodedDate)
+                print(decodedDate)
+                tokens = decodedDate.split('-')
+                """
+                Set the date value to full date is supplied, or to just month and year, or just year
+                Default to Jan, 1, 2020
+                """
+                day = 1
+                month = 1
+                year = 2020
+                if len(tokens) == 3:
+                    year = int(tokens[0])
+                    month = int(tokens[1])
+                    day = int(tokens[2])
+                elif len(tokens) == 2:
+                    year = int(tokens[0])
+                    month = int(tokens[1])
+                elif len(tokens) == 1:
+                    year = int(tokens[0])
```

Comment:
This seems a bit brittle. Are we specifically expecting omitted (and thus default) values? I think for simplicity we can assume that users have to input the same date format (especially considering the javascript library should be configured to output the same date format every time). 

Instead, I suggest using `strptime` and passing in the format we expect. This should make it more portable/easier to maintain. 

For instance, what happens here if the user enters `1-1-1`? I get an `OverflowError` which would be catastrophic (resulting in a 500 status page) in production. Better to reject these values as invalid from the start.

Reply:
Bill mentioned he'd like to see people have option to just filter on just month and year. So, yes that was the intention here. I'll look at strptime, thanks. I'll look at the widget rejecting invalid numbers. Any suggestions here @cantsin ?

Reply:
@cantsin From the datepicker, I've limited the date to minimum of 01/01/2020, so selection on the calendar is not available for the prior dates (years/months). But users can still type in, i'm looking to catch at the filter level.

Reply:
Datae filter has been revamped and is ready to be reviewed